src/status/signals/messages.nim

```diff
@@ -362,7 +369,36 @@ proc toMessage*(jsonMsg: JsonNode, pk: string): Message =
       signature: jsonMsg["commandParameters"]["signature"].getStr
     )
 
-  message.hasMention = concat(message.parsedText.map(t => t.children.filter(c => c.textType == "mention" and c.literal == pk))).len > 0
+    # This is kind of a workaround in case we're processing a transaction message. The reason for 
+    # that is a message where a recipient accepted to share his address with sender. In that message
+    # a recipient's public key is set as a "from" property of a "Message" object and we cannot 
+    # determine which of two users has initiated transaction actually. 
+    # 
+    # To overcome this we're checking if the "from" address from the "commandParameters" object of 
+    # the "Message" is contained as an address in the wallet of logged in user. If yes, means that
+    # currently logged in user has initiated a transaction (he is a sender), otherwise currently 
+    # logged in user is a recipient.
+    message.isCurrentUser = currentUserWalletContainsAddress(message.commandParameters.fromAddress)
```

Comment:
Maybe we could add an `if` to check if there is a `message.commandParameters.fromAddress` first, that way we don't call `currentUserWalletContainsAddress` with an empty sring.